ios/Exponent/Kernel/Views/EXAppViewController.m

```diff
@@ -305,6 +307,38 @@ - (void)errorViewDidSelectRetry:(EXErrorView *)errorView
 
 - (UIInterfaceOrientationMask)supportedInterfaceOrientations
 {
+  NSArray *legacy = @[@"33.0.0", @"34.0.0", @"35.0.0"];
+  NSString *sdkVersion = _appRecord.appLoader.manifest[@"sdkVersion"];
+  if ([legacy containsObject:sdkVersion]) {
+    return [self oldSupportedOrientationsLogic];
+  }
+  return [self newSupportedOrientationsLogic];
+}
+
+- (UIInterfaceOrientationMask)newSupportedOrientationsLogic {
```

Comment:
I think it would be clearer if we move this logic into `supportedInterfaceOrientations` and only leave the old implementation under `legacySupportedInterfaceOrientations` method. When you're introducing some legacy code, then it is always better to make it as easy as possible to remove the legacy code.